spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/ehcache/EhCacheAutoConfiguration.java

```diff
@@ -0,0 +1,68 @@
+package org.springframework.boot.autoconfigure.ehcache;
+
+import net.sf.ehcache.CacheManager;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cache.ehcache.EhCacheCacheManager;
+import org.springframework.cache.ehcache.EhCacheManagerUtils;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.util.Assert;
+
+import javax.annotation.PostConstruct;
+
+/**
+ * {@link org.springframework.boot.autoconfigure.EnableAutoConfiguration Auto-configuration} for EhCache.
+ *
+ * @author Eddú Meléndez
+ * @since 1.3.0
+ */
+@Configuration
+@ConditionalOnClass(CacheManager.class)
+public class EhCacheAutoConfiguration {
+
+	@Configuration
+	@ConditionalOnMissingBean(org.springframework.cache.CacheManager.class)
+	@EnableConfigurationProperties(EhCacheProperties.class)
+	public static class EhCacheManagerFactoryConfiguration {
+
+		@Autowired
+		private EhCacheProperties properties;
+
+		@Autowired
+		private ResourceLoader resourceLoader;
+
+		@PostConstruct
+		public void checkLocationExists() {
+			if (this.properties.isCheckLocation()) {
+				Assert.state(!this.properties.getLocation().isEmpty(),
+					"EhCache config file location is not configured");
+				boolean exists = hasLocation();
+				Assert.state(exists,
+					"Cannot find ehcache config file location in: " + this.properties.getLocation()
+						+ " (please add ehcache config file or check your EhCache configuration)");
+			}
+		}
+
+		private boolean hasLocation() {
+			String location = this.properties.getLocation();
+			if (this.resourceLoader.getResource(location).exists()) {
+				return true;
+			}
+			return false;
+		}
+
+		@Bean
+		public org.springframework.cache.CacheManager cacheManager() {
+			return new EhCacheCacheManager(EhCacheManagerUtils
+				.buildCacheManager(properties.getName(),
+					new ClassPathResource(properties.getLocation())));
+		}
```

Comment:
That makes the checkLocation thing a bit useless isn't it? We can translate this code by "if ehcache is on the classpath, then we try to create a config based on the specified location". 

So if you have ehcache on your classpath and no plan to use it (because you don't want any cache) then your application will not start because said file won't be found.

We need to put that in perspective with other caching infrastructure. Maybe this thing should be opt-in after all.